### PR TITLE
Add 'dagster code-server start' support as an option to the Helm chart

### PR DESCRIPTION
**Pull request by gibsondan**
Summary:
- make 'codeServerArgs' an option alongside 'dagsterApiGrpcArgs' that spins up the new entrypoint.

### CODE REVIEW

**Comment by alangenfeld** [comment]
what makes you confident enough to flip the default immediately? I would assume we want to have observed the proxy server in production environments a bit first

File: helm/dagster/values.yaml [hunk 1]
```diff
@@ -292,15 +292,13 @@ dagster-user-deployments:
...
+      # Ex: "dagster code-server start -m dagster_test.test_project.test_jobs.repo"
       # would translate to:
-      # dagsterApiGrpcArgs:
+      # codeServerArgs:
       #   - "-m"
       #   - "dagster_test.test_project.test_jobs.repo"
-      #   - "-a"
-      #   - "define_demo_execution_repo"
-      dagsterApiGrpcArgs:
+      codeServerArgs:
```
Comment (alangenfeld) [inline helm/dagster/values.yaml:301]: [1]

File: helm/dagster/charts/dagster-user-deployments/values.yaml [hunk 2]
```diff
@@ -31,15 +31,13 @@ deployments:
...
+    # Ex: "dagster code-server start -m dagster_test.test_project.test_jobs.repo"
     # would translate to:
-    # dagsterApiGrpcArgs:
+    # codeServerArgs:
     #   - "-m"
     #   - "dagster_test.test_project.test_jobs.repo"
-    #   - "-a"
-    #   - "define_demo_execution_repo"
-    dagsterApiGrpcArgs:
+    codeServerArgs:
```
Comment (alangenfeld) [inline helm/dagster/charts/dagster-user-deployments/values.yaml:40]: [1]

**Review by alangenfeld — COMMENTED** [review]
[1] I *think* its worth communicating the difference between `codeServerArgs` (reloadable vis proxy) and `dagsterApiGrpcArgs` (direct server). I can imagine someones managing a deployment may choose to save memory if they don't care about reload-ability for example.

**Comment by alangenfeld** [comment]
> I think having two ways to do things is confusing and the memory footprint should be pretty minimal (particularly since it's not loading any user code anymore).

I buy the argument against the added complexity. In my past explorations without optimizing for it we pull in a lot on import leading to a surprisingly high memory foot print before any user code is loaded (hundreds of MB).

I think some measurements or observations from running it in production would be convincing on what path to take on complete deprecation

**Comment by gibsondan** [comment]
"flipping the default" is slightly stronger language than what's actually happening, it wouldn't affect anybody that already has dagsterApiGrpcArgs set - it's more like 'flip the recommendation for new users coming to the Helm chart for the first time'. I think you're still right though, i'll do a less invasive change that makes it an option without it being the default recommendation

**Comment by gibsondan** [comment]
@alangenfeld i added a less aggro version that makes it an option but not the default - I was thinking i'd soft-launch it in the github issue with some instructions for how to turn it on before referencing it in the values.yaml at all to avoid confusion

**Review by alangenfeld — APPROVED** [review]
I think adding comments to values.yaml about it would be fine too but you call the ball

**Pull request merged**